### DISCUSSION TITLE
The type of audio used in the application.

### DISCUSSION THREAD

**Discussion by ma7moud3zim** [focustimerhq/FocusTimer#662] [Q&A] [answered]
The next Image is in the "notifications-capability.vala" file

![image](https://github.com/gnome-pomodoro/gnome-pomodoro/assets/72660571/e7e63d2e-5e92-49ee-abd6-d764b302fb45)

I want to know how the sound are played? and what the type of it [ mp3, wav, etc..].

If there are video or something that explain it I will be thankful. And may be I will try to contribute so anyone can add a mp3 audio file to the notfication.

**Comment by ma7moud3zim** [comment]
I tried to add wav audio and it works well. But I want to understand how it works maybe I can then edit the code so anyone can add mp3 audio file directly.

**Reply by kamilprusko** [reply, marked as answer]
Hi :)

The sounds are handled in plugins/sounds/sounds-plugin.vala, the `SoundManager` class.

For ticking sound it uses gstreamer. For notifications it uses gsound / libcanberra. libcanberra is more lightweight and supports caching + it follow system rules for notification sounds (volume / mute).

The list of supported types are hardcoded in plugins/sounds/sound-player.vala , `CanberraPlayer.get_supported_mime_types()`.

It could be that .mp3 is now supported.

**Reply by ma7moud3zim** [reply]
GitHub didn't notify me of this comment. I searched for the application to check it. :3

Thanks for explain. I will try to learn more about it.

**Comment by kamilprusko** [comment]
In `main` / 1.0-alpha it fallbacks to gstreamer now, so there's (almost) no file-format restrictions.
